app/controllers/ops_controller/ops_rbac.rb

```diff
@@ -153,19 +153,24 @@ def rbac_tenant_edit
   end
 
   def tenant_form_fields
-    tenant = Tenant.find_by_id(params[:id])
+    tenant      = Tenant.find_by_id(params[:id])
+    root_tenant = tenant.default?
+    use_config  = tenant.use_config_for_attributes
 
     render :json => {
-      :name        => tenant.name,
-      :description => tenant.description,
-      :default     => tenant.default?,
-      :divisible   => tenant.divisible
+      :name                      => root_tenant && use_config ? MiqServer.my_server.get_config("vmdb").config.fetch_path(:server, :company) : tenant.name,
+      :description               => tenant.description,
+      :default                   => root_tenant,
+      :divisible                 => tenant.divisible,
+      :use_config_for_attributes => tenant.use_config_for_attributes
     }
   end
 
   def tenant_set_record_vars(tenant)
-    tenant.name        = params[:name]
+    # there is no params[:name] when use_config_attributes is checked
+    tenant.name        = params[:name] if params[:name]
     tenant.description = params[:description]
+    tenant.use_config_for_attributes = params[:use_config_for_attributes] == "on" if tenant.root?
```

Comment:
I think you want to always set

tenant.use_config_for_attributes = tenant.root? && (params[:use_config_for_attributes] == "on")